File: 5-longest-palindromic-substring/longest-palindromic-substring.cpp
class Solution {
public:
    string helper(string str, int left, int right){

        while(left >= 0 && right < str.size() && str[left] == str[right]){
            left--;
            right++;
        }
        return str.substr(left+1, right-left-1);
    }
    string longestPalindrome(string s) {
        string res = s.substr(0,1);
        int n = s.size();
        for(int i = 0;i<n-1;i++){
            string odd,even;
            odd = helper(s,i,i);
            even = helper(s,i,i+1);

            if(odd.size() > res.size()){
                res = odd;
            }
            if(even.size() > res.size()){
                res = even;
            }
        }
        return res;
    }
};

File: 5-longest-palindromic-substring/longest-palindromic-substring.java
class Solution {
    public String helper(String s, int l, int r){
        while(l>=0 && r<s.length() && s.charAt(l) == s.charAt(r)){
            l--;
            r++;
        }
        return s.substring(l+1,r);
    }
    public String longestPalindrome(String s) {
        String result = "";
        for(int i = 0;i<s.length();i++){
            String odd = helper(s,i,i);
            String even = helper(s,i,i+1);
            if(odd.length() > result.length()){
                result = odd;
            }
            if(even.length() > result.length()){
                result = even;
            }
        }
        return result;
    }
}